### DISCUSSION TITLE
Pragmatic yet exhaustive UX/UI mockups

### DISCUSSION THREAD

**Discussion by warchimede** [MobileNativeFoundation/discussions#83] [General]
At France Télévisions, as a public service, we try to offer the best experience on as many devices as possible for our video streaming applications.
And, yet, we realize that our UX/UI mockups do not currently cover all possible cases.

We would like to come up with a pragmatic set of rules or guidelines to improve our UI mockups so that we can target the largest number of mobile devices, while also decreasing inconsistencies and the number of unsupported corner cases during development.

After discussing it among ourselves, here are our suggestions regarding the sizes for which we need a UI mockup :
- variant "small" (from 320 to 599) -> create a mockup for 320, since it is the smallest phone supported (iPhone 5)
- variant "medium" (from 600 to 899) -> create a mockup for 600, since, on Android, a device is considered as a tablet when its smallest dimension is 600
- variant "large" (900+) -> create a mockup for 900

Basically, for each variant, create a mockup for its smallest dimension.

Do you have any thoughts about this topic ?
Which processes or rules are in place in your teams ?

Thanks 🙏🏽

**Comment by gabriellanata** [comment]
Hello! Thanks for starting this thread! I'm an engineer in the Design Systems team at Lyft, we did some explorations on what would work best for us a few years ago. Note that our apps only support mobile phone sizes, so this exploration was mostly done for web. 

As a growing company with a large amount of different websites (internal and external) we realized that there was no magic rules that worked best for everyone. Each website served a different audience and it was best to optimize for their specific use cases vs a single rule for all. We broke it down into 5 possible sizes

![image](https://user-images.githubusercontent.com/585835/115082049-1f3b0780-9eba-11eb-9cec-ef53a4fb4e58.png)

And these are the breakpoints that they map to:

![image](https://user-images.githubusercontent.com/585835/115081976-fd418500-9eb9-11eb-89e5-ef760556c7f0.png)

In the end we decided on defining a set of breakpoints and allowing each team to decide what sizes they need to optimize for. For example, data from our lyft.com website shows traffic at less than 5% on devices at the M screen size. With this information and lack of layout change, the team decided to apply one breakpoint (<767 and >=768). Other services like our concierge website (used by hotels to help guests order Lyfts) was heavily used on tablets so the focus was different.

One additional note: We decided against rounding down breakpoints because that made designs have less available space with very little benefit. There are very few devices between 900 and 1023pt, so optimizing for those reduces the ability to optimize for devices at 1024pt, which there are a lot.

**Reply by warchimede** [reply]
Thank you Gabriel for your answer 😃
